test/conftest.py

```diff
@@ -23,5 +23,5 @@
 
 
 @pytest.fixture
-def tmpdir(tmpdir):
-    yield Path(str(tmpdir))
+def tmp_path(tmp_path):
```

Comment:
Optional: You could put this whole thing in an
```
if sys.version_info < (3, 6):
```
Block to very clearly show that this is only needed before python 3.6, because people generally don't look at blames for this kind of information. Maybe with an in-code comment on why this is needed, but for that a commit message can be sufficient.